tchannel-core/src/main/java/com/uber/tchannel/api/TChannel.java

```diff
@@ -174,69 +129,38 @@ public ChannelFuture listen() throws InterruptedException {
         return f;
     }
 
-    public void shutdown() throws InterruptedException {
-        this.peerManager.close();
-        this.bossGroup.shutdownGracefully();
-        this.childGroup.shutdownGracefully();
-    }
-
-    public <T, U> ListenableFuture<Response<T>> callThrift(
-            InetAddress host,
-            int port,
-            Request<U> request,
-            final Class<T> responseType
-    ) throws InterruptedException, TChannelError {
-        return callWithEncoding(host, port, request, responseType, ArgScheme.THRIFT);
-    }
-
-    public <T, U> ListenableFuture<Response<T>> callJSON(
-            InetAddress host,
-            int port,
-            Request<U> request,
-            final Class<T> responseType
-    ) throws InterruptedException, TChannelError {
-        return callWithEncoding(host, port, request, responseType, ArgScheme.JSON);
+    public SubChannel getSubChannel(String service) {
+        return subChannels.get(service);
     }
 
-    public ListenableFuture<RawResponse> call(
-            InetAddress host,
-            int port,
-            RawRequest request
-    ) throws InterruptedException, TChannelError {
-
-        // Set the ArgScheme as RAW if its not set
-        Map<String, String> transportHeaders = request.getTransportHeaders();
-        if (!transportHeaders.containsKey(TransportHeaders.ARG_SCHEME_KEY)) {
-            request.setTransportHeader(TransportHeaders.ARG_SCHEME_KEY, ArgScheme.RAW.getScheme());
+    // TODO: make sure makeSubChannel is called before listening ...
+    public SubChannel makeSubChannel(String service, Connection.Direction preferredDirection) {
+        SubChannel subChannel = getSubChannel(service);
+        if (subChannel == null) {
+            subChannel = new SubChannel(service, this, preferredDirection);
```

Comment:
This is fine. The only case when SubChannel needs to be GC'ed is in Hyperbahn implementation. Even in that case, it should be explicitly removed.